test/langtools/tools/javac/7166455/CheckACC_STRICTFlagOnclinitTest.java

```diff
@@ -83,18 +86,37 @@ private void run() throws IOException {
         }
     }
 
-    void check(String dir, String... fileNames) throws IOException{
+    void check(String dir, String... fileNames)
+        throws
+            IOException,
+            ConstantPoolException,
+            Descriptor.InvalidDescriptor {
         for (String fileName : fileNames) {
-            ClassModel classFileToCheck = Classfile.of().parse(new File(dir, fileName).toPath());
+            ClassFile classFileToCheck = ClassFile.read(new File(dir, fileName));
 
-            for (MethodModel method : classFileToCheck.methods()) {
-                if ((method.flags().flagsMask() & Classfile.ACC_STRICT) == 0) {
+            for (Method method : classFileToCheck.methods) {
+                if ((method.access_flags.flags & ACC_STRICT) == 0) {
                     errors.add(String.format(offendingMethodErrorMessage,
-                            method.methodName().stringValue(),
-                            classFileToCheck.thisClass().asInternalName()));
+                            method.getName(classFileToCheck.constant_pool),
+                            classFileToCheck.getName()));
                 }
             }
         }
     }
 
+// this version of the code can be used when ClassFile API in not in a preview
```

Comment:
Do you want to keep these commented lines?

Reply:
That said, if tests with such usages are not many, we could also consider adding enable-preview to the tests, and just use the new API.

Reply:
Yes, that is proposed temporary solution to do not loose already converted code while ClassFile API is in preview.